api/main.py

```diff
@@ -368,3 +371,120 @@ class parallelItem(BaseModel):
+                "selected": selected,
+            },
+        )
+        category_list = {}
+        for category in query_full_category_list.result:
+            category_list.update(category)
+
+        # check if the search is for a catagory (i.e. T06) or for a collection (i.e. Tengyur)
+        if searchtype == "category":
+            filecount = get_query_files_per_category(language+"_"+searchterm)
+
+            for filename in filecount:
+                parallel_count = filename["totallengthcount"]
+                for categoryname in category_list.keys():
+                    if categoryname in parallel_count.keys():
+                        total_parallel_count.append([filename["filename"],"R_"+categoryname+" "+category_list[categoryname],parallel_count[categoryname]])
+                    else:
+                        total_parallel_count.append([filename["filename"],"R_"+categoryname+" "+category_list[categoryname],0])
+
+        # if the search is for a collection, a list of categories for that collection
+        # is iterated over and the results for each file added.
+        elif searchtype == "collection":
+            query_collection_list = db.AQLQuery(
+                query=query_categories_per_collection,
+                bindVars={
+                    "searchterm": language+"_"+searchterm,
+                    "language": language
+                },
+            )
+
+            for cat, catname in query_collection_list.result[0].items():
+                filecount = get_query_files_per_category(language+"_"+cat)
+
+                total_parlist = {}
+                for filename in filecount:
+                    parallel_count = filename["totallengthcount"]
+                    for categoryname in category_list.keys():
+                        if categoryname not in total_parlist.keys():
+                            if categoryname not in parallel_count.keys():
+                                total_parlist[categoryname] = 0
+                            else:
+                                total_parlist[categoryname] = parallel_count[categoryname]
+                        elif categoryname in parallel_count.keys():
+                                total_parlist[categoryname] += parallel_count[categoryname]
+
+                for key, value in total_parlist.items():
+                    total_parallel_count.append(["L_"+cat+" "+catname, "R_"+key+" "+category_list[key], value])
+
+        return { "graphdata" : total_parallel_count }
+
+    except DocumentNotFoundError as e:
+        print(e)
+        raise HTTPException(status_code=404, detail="Item not found")
+    except AQLQueryError as e:
+        print("AQLQueryError: ", e)
+        raise HTTPException(status_code=400, detail=e.errors)
+    except KeyError as e:
+        print("KeyError: ", e)
+        raise HTTPException(status_code=400)
+
+
+# supporting function to return a list of files in the respective category
+def get_query_files_per_category(
```

Comment:
since this is not an endpoint, it doesn't have to be in this file. Actually, maybe we can start creating facades for the DB queries, similar to `actions` in the frontend? Something like:

`db_actions.py`
```python
async def get_files_per_category_from_db(search_term):
	db = get_db()
    files_per_category = db.AQLQuery(
        query=query_files_per_category,
        batchSize=100000,
        bindVars={
            "searchterm": search_term,
        },
    )

    return files_per_category.result
```

This way we could keep the endpoint code smaller.

Reply:
If I do this, which imports other than "query_files_par_category" would I need. Any of these?
```
from .utils import get_language_from_filename, get_regex_test, get_future_date
from .db_connection import get_collection, get_db
from fastapi import FastAPI, HTTPException, Query
from pyArango.theExceptions import DocumentNotFoundError, AQLQueryError
from starlette.middleware.cors import CORSMiddleware
from starlette.responses import Response
from pydantic import BaseModel
```

Reply:
Done.

Reply:
Which editor do you use @ayya-vimala ? Mine (PyCharm) automatically suggests imports, VsCode should too I think.